sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -122,6 +130,20 @@ class SingleStatementExec(
    */
   var isExecuted = false
 
+  /**
+   * Plan with named parameters.
+   */
+  lazy val resolvedPlan: LogicalPlan = {
+    if (args.nonEmpty) {
+      NameParameterizedQuery(parsedPlan, args)
+    } else {
+      parsedPlan
+    }
+  }
+
+  /** Statement is result if it is a SELECT query, and it is not in control flow condition */
+  override def isResult: Boolean = parsedPlan.isInstanceOf[Project] && !isExecuted
```

Comment:
With SQL pipe syntax, the root node may not always be `Project`. I think it's simpler to let the caller side check `df.logicalPlan`. If it's `CommandResult`, then it's not a query.

Reply:
IIUC, `SqlScriptExecution` should be responsible to check if current DataFrame.logicalPlan  is `Project` or not, instead of `SingleStatementExec` based on it's logical plan?